tests/DotNetLightning.Core.Tests/TransactionTests.fs

```diff
@@ -326,4 +329,497 @@ let testList = testList "transaction tests" [
+                    let nextLocalPerCommitmentSecretsRes =
+                        remoteRemotePerCommitmentSecrets.InsertPerCommitmentSecret
+                            currentCommitmentNumber
+                            currentPerCommitmentSecret
+                    addKeys
+                        (Result.deref nextLocalPerCommitmentSecretsRes)
+                        (currentCommitmentNumber.NextCommitment())
+            addKeys (PerCommitmentSecretStore()) CommitmentNumber.FirstCommitment
+
+        let remoteRemoteCommit = {
+            Index = commitmentNumber
+            Spec = localCommitmentSpec
+            TxId = TxId <| localCommitmentTx.GetHash()
+            RemotePerCommitmentPoint = perCommitmentPoint
+        }
+
+        let remoteRemoteParams = {
+            DustLimitSatoshis = localParams.DustLimitSatoshis
+            MaxHTLCValueInFlightMSat = localParams.MaxHTLCValueInFlightMSat
+            ChannelReserveSatoshis = localParams.ChannelReserveSatoshis
+            HTLCMinimumMSat = localParams.HTLCMinimumMSat
+            ToSelfDelay = localParams.ToSelfDelay
+            MaxAcceptedHTLCs = localParams.MaxAcceptedHTLCs
+            Features = localParams.Features
+        }
+
+        let staticRemoteChannelConfig : StaticChannelConfig = 
+            {
+                FundingScriptCoin = fundingScriptCoin
+                AnnounceChannel = false
+                RemoteNodeId = localNodeMasterPrivKey.NodeId()
+                Network = Network.RegTest
+                IsFunder = false
+                FundingTxMinimumDepth = BlockHeightOffset32 6u
+                LocalStaticShutdownScriptPubKey = None
+                RemoteStaticShutdownScriptPubKey = None
+                LocalParams = remoteLocalParam
+                RemoteParams = remoteRemoteParams
+                RemoteChannelPubKeys = localChannelPubKeys
+            }
+
+        let remoteLocalCommit : LocalCommit =
+            {
+                Index = commitmentNumber
+                Spec = remoteCommitmentSpec
+                PublishableTxs = {
+                    CommitTx = FinalizedTx remoteCommitmentTx
+                    HTLCTxs = List.Empty
+                }
+                PendingHTLCSuccessTxs = List.Empty
+            }
+
+        let remoteSavedChannelState : SavedChannelState = {
+            StaticChannelConfig = staticRemoteChannelConfig
+            RemotePerCommitmentSecrets = remoteRemotePerCommitmentSecrets
+            ShortChannelId = None
+            LocalCommit = remoteLocalCommit
+            RemoteCommit = remoteRemoteCommit 
+            LocalChanges = LocalChanges.Zero
+            RemoteChanges = RemoteChanges.Zero
+        }
+
+        let validateClosingResult (expectedAmount: Money) (errorMsg: string) (result: ClosingHelpers.ClosingResult) =
+            let dest =
+                let remoteDestPrivKey = new Key()
+                let remoteDestPubKey = remoteDestPrivKey.PubKey
```

Comment:
nit: you don't have to create variable `remoteDestPubKey` and just use`remoteDestPrivKey.PubKey.WitHash.ScriptPubKey` directly in case like this.
The only case you should assign a new variable like this is to give a new information to the reader by its name.
e.g. 

```fsharp
// good example
let pubkeyToUseInMultisig = myPrivKey.PubKey // it is not self-evident from the right hand that this key is going to be used for multisig

// bad example
let pubkeyToUseInMultisig = myPrivKeyForMultisig.PubKey // no new information
```

Reply:
Fixed

Reply:
> P.S: all these codes are taken from the same TransactionTests file so I didn't really pay more attention to this sort of stuff

I cannot believe I wrote code like this by myself ...

Reply:
Yes, it was probably missed in previous reviews 👍🏻 
https://github.com/joemphilips/DotNetLightning/blob/master/tests/DotNetLightning.Core.Tests/TransactionTests.fs#L159